src/crawler.ts

```diff
@@ -152,7 +152,7 @@ export class ResolveCrawler implements Types.ICrawler {
 
         // if we are partially resolving, we need to follow refs (since they might point outside of our initial target object tree)
         // only need to initiate when top of pointer stack
-        if (this.jsonPointer) {
+        if (this.jsonPointer && !this.pointerGraph.dependantsOf(parentPointer).includes(targetPointer)) {
```

Comment:
Marc, it seems like I'll need to revert to `pointerStack.includes(targetPointer)`, as this reliably fixes the issue for the larger Studio spec and the test I added.
I'll include that larger spec too, to make sure we don't regress in future

Reply:
That sounds good to me - it's a fine fix IMHO.

Reply:
Done. Included a test that was failing with the above change.